src/components/Buttons.js

```diff
@@ -0,0 +1,19 @@
+import React from "react";
+
+function Buttons({name, fruit,setFruit}) {
+ return(
+    <div className="button">
+        <h1>{name}</h1>
+        <button
+            disabled={fruit === 0}
+            onClick={()=> setFruit(fruit -1)}
```

Comment:
Wellicht gaat dit nu iets te ver, maar doordat je `fruit` gebruikt in deze functie zorgt dit bij een verandering van de waarde van `fruit` dat deze functie telkens voor een onnodige rerender zorgt. Je kan dit tegen gaan door het volgende te doen:
```suggestion
            onClick={()=> setFruit(curr => curr -1)}
```

`curr` is in dit voorbeeld de huidige waarde van `fruit`, dezelfde waarde dus. Het werkt hetzelfde, maar zorgt niet voor een onnodige rerender.

Het is niet heel erg als je dat rerender-gebeuren nog niet echt doorhebt of kan onthouden, maar ik wil er je wel op wijzen 😉 . In essentie gaat het er dus om dat wanneer je iets met die set-functie wil doen van de useState, dan kun je de huidige waarde op deze manier gebruiken om een nieuwe waarde te setten.